### PR TITLE
Document accessing multiple azure storage accounts in single catalog

### PR DESCRIPTION
**Pull request by anusudarsan**
## Description

For https://github.com/trinodb/trino/issues/23538

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
## Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 1]
```diff
@@ -91,3 +91,14 @@ Storage:
 * - `azure.oauth.secret`
   - A OAuth 2.0 client secret for the service principal.
 :::
+
+## Access multiple storage accounts
+
+To allow Trino to access multiple Azure storage accounts from a single
+catalog configuration, you can use [](azure-oauth-authentication) with a
+service principal. First, create a service principal in Azure Active
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:99]:
```suggestion
service principal:

First, create a service principal in Azure Active
```

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 2]
```diff
@@ -91,3 +91,14 @@ Storage:
...
+Directory (App registrations) and save the client secret. Then, grant the
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:100]: What is this "(App registration)" trying to imply?

Comment (mosabua) [reply docs/src/main/sphinx/object-storage/file-system-azure.md:100]: Also .. potentially the "Then.. " could be a new paragraph or you could use numbered steps

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 3]
```diff
@@ -91,3 +91,14 @@ Storage:
...
+service principal access by assigning appropriate roles, such as Storage
+Blob Data Contributor, to each storage account. This is done via the Access
+Control (IAM) section, where you add role assignments and select the service
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:103]: Again .. what is "Access Control (IAM)" .. if this is a title or screen or table name in the Ui.. use bold

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 4]
```diff
@@ -91,3 +91,14 @@ Storage:
...
 :::
+
+## Access multiple storage accounts
+
+To allow Trino to access multiple Azure storage accounts from a single
+catalog configuration, you can use [](azure-oauth-authentication) with a
+service principal. First, create a service principal in Azure Active
+Directory (App registrations) and save the client secret. Then, grant the
+service principal access by assigning appropriate roles, such as Storage
+Blob Data Contributor, to each storage account. This is done via the Access
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:102]:
```suggestion
Blob Data Contributor, to each storage account. Use the Access
```

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 5]
```diff
@@ -91,3 +91,14 @@ Storage:
...
   - A OAuth 2.0 client secret for the service principal.
 :::
+
+## Access multiple storage accounts
+
+To allow Trino to access multiple Azure storage accounts from a single
+catalog configuration, you can use [](azure-oauth-authentication) with a
+service principal. First, create a service principal in Azure Active
+Directory (App registrations) and save the client secret. Then, grant the
+service principal access by assigning appropriate roles, such as Storage
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:101]: I think Storage Blob Data Contributor is an actual name of a role .. so please bold it

**Review by mosabua — CHANGES_REQUESTED** [review]

**Comment by mosabua** [comment]
Also .. just to be sure .. this page is generic for all table formats.. so the assumption is that the approach works for all of them. If not we need to clarify

**Review by anusudarsan — COMMENTED** [review]
@mosabua addressed comments. Also added a bit more details. please re-review

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 6]
```diff
@@ -91,3 +91,26 @@ Storage:
...
+
+## Access multiple storage accounts
+
+To allow Trino to access multiple Azure storage accounts from a single
+catalog configuration, you can use [](azure-oauth-authentication) with
+an Azure service principal. The following steps describe how to create
+a service principal in Azure and assign an IAM role granting access to the
+storage accounts:
+
+- Create a service principal in Azure Active Directory (using Azure
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:103]:
```suggestion
- Create a service principal in Azure Active Directory using Azure
```

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 7]
```diff
@@ -91,3 +91,26 @@ Storage:
...
+  `App Registrations`) and save the client secret.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:104]:
```suggestion
  **App Registrations** and save the client secret.
```

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 8]
```diff
@@ -91,3 +91,26 @@ Storage:
...
+- You can assign this service principal to have access to the storage
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:105]:
```suggestion
- You can assign access to the storage
```

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 9]
```diff
@@ -91,3 +91,26 @@ Storage:
...
+  account(s) from the account's `Access Control (IAM)` section. You can add
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:106]:
```suggestion
  accounts this service principal from the account's **Access Control (IAM)** section. You can add
```

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 10]
```diff
@@ -91,3 +91,26 @@ Storage:
...
+  `Role Assignments` and select appropriate roles, such
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:107]:
```suggestion
  **Role Assignments** and select appropriate roles, such
```

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 11]
```diff
@@ -91,3 +91,26 @@ Storage:
...
+  as `Storage Blob Data Contributor`.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:108]:
```suggestion
  as **Storage Blob Data Contributor**.
```

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 12]
```diff
@@ -91,3 +91,26 @@ Storage:
...
+- Assign access using the option `User, group, or service principal` and
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:109]:
```suggestion
- Assign access using the option **User, group, or service principal** and
```

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 13]
```diff
@@ -91,3 +91,26 @@ Storage:
...
+  select the service principal created. Save to finalize the role
+  assignment.
+
+ Once you create the service principal and configure the storage account(s)
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:113]:
```suggestion
 Once you create the service principal and configure the storage accounts
```

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 14]
```diff
@@ -91,3 +91,26 @@ Storage:
...
+ use the `Client ID`, `Secret` and `Tenant ID` from the application
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:114]:
```suggestion
 use the **Client ID**, **Secret**, and **Tenant ID** values from the application
```

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 15]
```diff
@@ -91,3 +91,26 @@ Storage:
...
+ registration, to configure the catalog using properties from
+ [](azure-oauth-authentication).
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:116]: empty line after code

**Review by mosabua — CHANGES_REQUESTED** [review]
Sorry.. more suggestions .. nearly there.

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 16]
```diff
@@ -91,3 +91,25 @@ Storage:
...
+  **App Registrations** and save the client secret.
+- Assign access to the storage accounts from the account's
+  **Access Control (IAM)** section. You can add **Role Assignments** and
+  select appropriate roles, such as **Storage Blob Data Contributor**.
+- Assign access using the option **User, group, or service principal** and
+  select the service principal created. Save to finalize the role
+  assignment.
+
+ Once you create the service principal and configure the storage accounts
+ use the **Client ID**, **Secret** and **Tenant ID** from the application
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/file-system-azure.md:113]:
```suggestion
 use the **Client ID**, **Secret** and **Tenant ID** values from the application
```

**Review by mosabua — COMMENTED** [review]
One nit and a final update of the commit message please .. then its good to go

https://trino.io/development/process#pull-request-and-commit-guidelines-

**Review by mosabua — APPROVED** [review]

**Pull request merged**